### PR TITLE
DOCSP-40795: GridFS

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40795
Staging - https://deploy-preview-67--docs-cpp.netlify.app/write/gridfs/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by mongoKart — APPROVED** [review]
LGTM w/ some small suggestions!

File: source/write/gridfs.txt [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+- ``chunks`` collection, which stores the binary file chunks
+- ``files`` collection, which stores the file metadata
+
+The driver creates the GridFS bucket, if it doesn't exist, when you perform
+the first write operation. The bucket contains the preceding collections
+prefixed with the default bucket name ``fs``, unless you specify a different
+name. To ensure efficient retrieval of the files and related metadata, the driver
+also creates an index on each collection if they don't exist and when the
+bucket is empty.
```
[the remark this answers is not in the document]
Comment (norareidy) [reply source/write/gridfs.txt]: It's a little unclear - I reworded based on the Server manual info

File: .vscode/settings.json
Comment (mongoKart) [inline .vscode/settings.json:1]: (be sure to keep this out of the merge)

File: source/write/gridfs.txt [hunk 2]
```diff
@@ -43,12 +43,12 @@ defined in the GridFS specification:
...
-The driver creates the GridFS bucket, if it doesn't exist, when you perform
-the first write operation. The bucket contains the preceding collections
+The driver creates the GridFS bucket, if it doesn't exist, when you first
+write data to it. The bucket contains the preceding collections
 prefixed with the default bucket name ``fs``, unless you specify a different
 name. To ensure efficient retrieval of the files and related metadata, the driver
-also creates an index on each collection if they don't exist and when the
-bucket is empty. 
+creates an index on each collection. The driver ensures that these indexes exist
+before performing read and write operations on the GridFS bucket.
```
Comment (mongoKart) [inline source/write/gridfs.txt:51]: 💯

**Pull request merged**